src/viam/components/resource_manager.py

```diff
@@ -64,6 +74,9 @@ def get_component(self, of_type: Type[ResourceType], name: ResourceName) -> Reso
         if component and isinstance(component, of_type):
             return component
 
+        if name.name in self._short_to_long_name and len(self._short_to_long_name[name.name]) == 1:
```

Comment:
Ok this makes me think we shouldn't be using a `List` at all to store the `ResourceName`s. Because if there is a collision (multiple of the same short names), then we store the additional ResourceNames. But down here, if the list is greater than 1, we ignore the short name. Then what's the point of storing it?

I think that you if you encounter a collision when creating the dictionary (line 47), then you should simply delete the entry from the dictionary itself and pretend like it was never there.

Reply:
we should store something because if there were 3 of the same name, we also shouldn't have an entry. We could go back and delete the entry at the end, but we shouldn't delete during the process of creating the dict.

Reply:
Big brain

Reply:
I was also wondering what the best method for this was, since I was thinking the same thing as Cheuk. I was tempted to delete the multiple entries at the end of `ResourceManager.__init__()`, but this would make it so that when `register()` is called separately elsewhere, then `_short_to_long_name()` will no longer be accurate.
Is there another smarter way to do this?

Reply:
we could just make `register` an internal method `_register` and hope people don't register anything on their own. We can't guarantee anything in Python haha.
The Go SDK implementation uses a second map (dupMap) which tracks the keys to delete at the end, so that we can make  `_short_to_long_name` `Dict[str,ResourceName]`, which might look and feel better.
I think either works

Reply:
I definitely agree that `Dict[str, ResourceName]` looks way better, but we already use `register` elsewhere, in `RobotClient.refresh()`.

Reply:
I definitely agree that Dict[str, ResourceName] looks way better, but we already use register elsewhere, in RobotClient.refresh().

Reply:
oh ya I missed that